node/src/main/scala/coop/rchain/node/diagnostics/StoreMetrics.scala

```diff
@@ -46,11 +46,16 @@ object StoreMetrics extends StoreMetricsInstances {
         g("total-size-on-disk", storeSize.totalSizeOnDisk),
         g("rspace-size-on-disk", storeSize.rspaceSizeOnDisk),
         g("rspace-data-entries", storeSize.rspaceDataEntries),
-        pc("rspace-consumes", storeSize.rspaceConsumesCount),
-        pc("rspace-produces", storeSize.rspaceProducesCount),
-        cm("rspace-consumes-COMM", storeSize.rspaceConsumesCommCount),
-        cm("rspace-produces-COMM", storeSize.rspaceProducesCommCount),
-        cm("rspace-install-COMM", storeSize.rspaceInstallCommCount)
+        pc("rspace-consumes", storeSize.rspace.flatMap(_.consumes)),
+        pc("rspace-produces", storeSize.rspace.flatMap(_.produces)),
+        cm("rspace-consumes-COMM", storeSize.rspace.flatMap(_.consumesComm)),
+        cm("rspace-produces-COMM", storeSize.rspace.flatMap(_.producesComm)),
+        cm("rspace-install-COMM", storeSize.rspace.flatMap(_.installComm)),
+        pc("replayrspace-consumes", storeSize.replayRSpace.flatMap(_.consumes)),
+        pc("replayrspace-produces", storeSize.replayRSpace.flatMap(_.produces)),
+        cm("replayrspace-consumes-COMM", storeSize.replayRSpace.flatMap(_.consumesComm)),
+        cm("replayrspace-produces-COMM", storeSize.replayRSpace.flatMap(_.producesComm)),
+        cm("replayrspace-install-COMM", storeSize.replayRSpace.flatMap(_.installComm))
```

Comment:
This looks very repetitive, you could use lens-es for that. If you generate a `Lens[RSpace|ReplayRSpace, Consumes|ConsumesComm|…]` you can do `consumesCommLens.get(replayRspace)` , `consumesCommLens.get(rspace)` etc. Decision is yours.

Reply:
Thanks, I'll have a look at this

Reply:
It would make sense if we had some nested case classes here, but we don't. I'm going to extract a helper function though